plugin/src/main/java/com/redhat/jenkins/nodesharingfrontend/Api.java

```diff
@@ -96,6 +99,13 @@ public Api(String OrchestratorUrl) {
         base = client.target(OrchestratorUrl);
     }
 
+    private SharedNodeCloud getCloud() {
+        if (cloud == null) {
+            throw new ActionFailed.CommunicationError("Cloud isn't set!");
```

Comment:
Is there a reason to ever construct an `Api` object without it? Provided it is not, let's enforce that in constructor.

Reply:
Ok, but we need a getApi() for testing purposes anyway...

Reply:
BTW in that case we need to deal with Interrupted ex. in the constructor... how we should handle it there? Set api var to null?